### PR TITLE
Fix EZP-24890: Delayed indexing should fail gracefully on trashed / broken node (alt)

### PR DESCRIPTION
**Pull request by peterkeung**
An alternative to https://github.com/ezsystems/ezpublish-legacy/pull/1204 with a simpler diff: duplicating the DB commands.

### CODE REVIEW

**Comment by emodric** [comment]
@peterkeung You have pa permission change in your diff (100644 -> 100755). I don't think this is required since this is a cronjob script, executed through runcronjobs.php, not directly, so no need for execute permissions.

File: cronjobs/indexcontent.php [hunk 1]
```diff
@@ -62,22 +62,24 @@
                     if ( !( $node instanceof eZContentObjectTreeNode ) )
                     {
                         $cli->error( "An error occured while trying fetching node $nodeId" );
+                        $db->query( "DELETE FROM ezpending_actions WHERE action = '$action' AND param = '$objectID'" );
+                        $db->commit();
```
Comment (joaoinacio) [inline cronjobs/indexcontent.php]: Won't this db `commit()` potentially introduce a transaction counter mismatch somewhere?

Comment (peterkeung) [inline cronjobs/indexcontent.php]: continue() brings us into a new foreach loop, which then starts a new transaction with $db->begin(), right?

**Comment by peterkeung** [comment]
@emodric Good catch!  I completely missed that.  I've removed that file permission change now.

**Comment by andrerom** [comment]
Besides the unrelated offset/limit variable renaming which should be removed;  +1 *(also for backport)*

**Comment by peterkeung** [comment]
Variable clash fix has been removed now. We can create a separate ticket for that.

**Comment by joaoinacio** [comment]
+1

**Comment by yannickroger** [comment]
+1
@peterkeung can you do the rename mentioned by @joaoinacio and ping me, I'll merge it asap.

**Comment by yannickroger** [comment]
@peterkeung @joaoinacio I will do the rename right now.

**Comment by yannickroger** [comment]
My bad you did it @peterkeung

**Comment by yannickroger** [comment]
Thanks @peterkeung

**Comment by yannickroger** [comment]
You did not do it… sorry… I added a note of the commit. They are duplicates so it's not too bad anyways.

**Pull request merged**